_posts/2015-09-18-Cocoa-Architecture-Dependencies.markdown

```diff
@@ -0,0 +1,129 @@
+
+#### Focus
+
+Ideally you're bringing in a dependency for one specific task. Something like [ObjectiveSugar](https://cocoapods.org/pods/ObjectiveSugar) is a great example of a small focused library. I studied it's API when we first integrated it, and it's not really changed at all since. The library authors have done a great job of ensuring that Objective Sugar stays on-topic.
+
+#### Community Relationships
+
+The amount of energy you have to put into using certain dependencies, in an engaging way like we do, also means you have to deal with people. This is often the case with OSS, so we’re used to that and quite good at it. Sometimes, however, people and their opinions don’t match, it’s a fact of life, and in these cases it can be worth it to choose to not use a certain dependency, if not strictly needed.
+
+One such example has been ReactiveCocoa. While they are completely in their right to do whatever they feel they need to do, we feel that we’ve wasted energy on supporting their dependencies in CocoaPods that took away from our already constraining time-budget. It consumes and creates energy that we don't want to be associated with, and makes us want to engage less with the community at whole, which is really not something we want. So that, combined with our opinion on the framework itself (as aforementioned) means we’re looking at alternatives.
+
+This contrasts sharply with working on Fastlane, where we're willing to take the dependency graph in part because working with [Felix Krause](https://krausefx.com) is such a pleasure. We've sent PRs,  helped out on [documentation](https://github.com/KrauseFx/fastlane/pull/173) and provided advice on how we would/are using Fastlane. I actively feel guilty that we're still not deploying to the App Store using Fastlane because of this relationship.
+
+#### Project Maturity
+
+We got burned by working with Swift too early, but coming back at Swift 2.0 feels good. We just started introducing Swift into our Objective-C codebases, as it's looking like the tooling has matured.
+
+We spent a good chunk of time over the last 6 months discussing and testing out React-Native as an approach for building apps. It's turning into an amazing platform, but for us it's not mature enough to start building apps with it as a foundation.
+
+## Not All Dependencies are Third Party
+
+We ship a lot of our internal code as Pods. The rubric we use for deciding on when to externalise code is "would we want this in more than 2 apps" at the minute. This is an easy choice for libraries as fundamental as a fonts, or colours schemes. It becomes a more nuanced choice when it comes to a class or two, or some functions.
```

Comment:
"at the minute" -> "at the moment" ?

"a class or two" -> link to forgeries maybe?